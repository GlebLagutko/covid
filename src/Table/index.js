import React, {useEffect, useState} from "react";
import Select from "../Select";
import SelectPeriod from "../SelectPeriod";


function Table() {
  let [currentCountry, setCountry] = useState('Belarus');
  let [cases, setCases] = useState(0);
  let [deaths, setDeaths] = useState(0);
  let [recovered, setRecovered] = useState(0);
  let [period, setPeriod] = useState('Year');

  function fetchData() {
    let url = '';
    if (currentCountry === 'All') {
      url = `https://disease.sh/v3/covid-19/${currentCountry.toLowerCase()}`;
    } else {
      url = `https://disease.sh/v3/covid-19/countries/${currentCountry.toLowerCase()}`;
    }

    fetch(url).then(response => response.json()).then(data => {

      if (period === 'Year') {
        setCases(data.cases);
        setDeaths(data.deaths);
        setRecovered(data.recovered);
      } else {
        console.log("cases" + data.todayCases)
        setCases(data.todayCases);
        setDeaths(data.todayDeaths);
        setRecovered(data.todayRecovered);
      }
    });
  }

  useEffect(fetchData, [currentCountry, period]);

  const updateCountry = (currentCountry) => {
    setCountry(currentCountry);
  };
  const updatePeriod = (period) => {
    setPeriod(period);
  };

  return (
    <div className='table '>
      <Select country={currentCountry} setCountry={updateCountry}/>
      <SelectPeriod period={period} setPeriod={updatePeriod}/>
      <div className='table-cell'>{cases}<span className='text-table'> cases</span></div>
      <div className='table-cell deaths'>{deaths}<span className='text-table'> deaths</span></div>
      <div className='table-cell recovered'>{recovered}<span className='text-table'> recovered</span></div>
    </div>
  )
}

export default Table;