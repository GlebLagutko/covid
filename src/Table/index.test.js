import Table from '.';
import React from "react";
import {render} from '@testing-library/react';
import {rest} from 'msw'
import {setupServer} from 'msw/node'

describe('Table', () => {

  const data = {cases: 1000, recovered: 0, deaths: 10};
  const server = setupServer(
    rest.get('https://disease.sh/v3/covid-19/countries/belarus', (req, res, ctx) => {
      return res(ctx.json(data))
    }),
  );

  beforeAll(() => server.listen());
  afterAll(() => server.close());
  afterEach(() => server.resetHandlers());

  it('renders correctly', async () => {

    const {container, findByText} = render(<Table/>);

    expect(await findByText("1000")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

});