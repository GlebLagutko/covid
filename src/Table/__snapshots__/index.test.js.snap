// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<div>
  <div
    class="table "
  >
    <select>
      <option
        value="All"
      >
        All
      </option>
    </select>
    <select>
      <option
        value="Year"
      >
        Year
      </option>
      )
      <option
        value="OneDay"
      >
        OneDay
      </option>
      )
    </select>
    <div
      class="table-cell"
    >
      1000
      <span
        class="text-table"
      >
        cases
      </span>
    </div>
    <div
      class="table-cell deaths"
    >
      10
      <span
        class="text-table"
      >
        deaths
      </span>
    </div>
    <div
      class="table-cell recovered"
    >
      0
      <span
        class="text-table"
      >
        recovered
      </span>
    </div>
  </div>
</div>
`;
